src/components/CookieConsentBanner.js

```diff
@@ -0,0 +1,185 @@
+import React from 'react'
+import ReactGA from 'react-ga'
+import Cookies from 'js-cookie'
+
+import {
+  c_PRIMARY_TEXT,
+  c_SECONDARY_TEXT,
+  breakpoint,
+  c_COOKIE_BACKGROUND,
+  c_COOKIE_BUTTON,
+  c_COOKIE_BUTTON_HOVER,
+  GUTTER_PX,
+} from '../theme'
+import Text from './Text'
+import VerticalSpacing from './VerticalSpacing'
+import Heading from './Heading'
+
+const Button = ({ onClick, className, children, open }) => (
+  <button type="button" onClick={onClick} className={className}>
+    {children}
+  </button>
+)
+
+const ACCEPTANCE_COOKIE = 'neontribe-cookies-accept'
+const NEONTRIBE_GA = 'UA-30970110-1'
+
+const startGA = () => {
+  ReactGA.initialize(NEONTRIBE_GA)
+}
+
+const stopGA = () => {
+  ReactGA.ga('remove')
+  Cookies.remove('_ga')
+  Cookies.remove('_gid')
+  Cookies.remove('_gat')
+}
+
+const CookieConsentBanner = () => {
+  const [isBannerOpen, setIsBannerOpen] = React.useState(false)
+  const [hasTrackedView, setHasTrackedView] = React.useState(false)
+  const [acceptanceCookie, setAcceptanceCookie] = React.useState(
+    Cookies.get(ACCEPTANCE_COOKIE)
+  )
+
+  React.useEffect(() => {
+    if (acceptanceCookie === 'true') {
+      startGA()
+    }
+
+    if (acceptanceCookie !== 'true') {
+      stopGA()
+    }
+  }, [acceptanceCookie])
+
+  React.useEffect(() => {
+    if (window.ga && !hasTrackedView) {
+      ReactGA.pageview(window.location.pathname + window.location.search)
+      setHasTrackedView(true)
+    }
+  }, [hasTrackedView, acceptanceCookie])
+
+  const updateAcceptanceCookie = () => {
+    setIsBannerOpen(false)
+    setAcceptanceCookie(Cookies.get(ACCEPTANCE_COOKIE))
+  }
+
+  const acceptCookies = () => {
+    Cookies.set(ACCEPTANCE_COOKIE, true, { expires: 28 * 3 })
+    updateAcceptanceCookie()
+  }
+
+  const declineCookies = () => {
+    Cookies.set(ACCEPTANCE_COOKIE, false)
```

Comment:
Does this not violate the request to not store cookies by storing a cookie?

Reply:
We think it comes under strictly necessary exemptions, with sufficient copy:
https://ico.org.uk/for-organisations/guide-to-pecr/guidance-on-the-use-of-cookies-and-similar-technologies/how-do-we-comply-with-the-cookie-rules/

![image](https://user-images.githubusercontent.com/32434620/83266913-65607d00-a1bb-11ea-893a-285aef60fca7.png)

Reply:
I guess this is very good explanation, I assume I can resolve this then.